pkg/ebpf/tracee.go

```diff
@@ -869,20 +741,20 @@ func (t *Tracee) getUnavKsymsPerEvtID() map[events.ID][]string {
 // validateKallsymsDependencies load all symbols required by events dependencies
 // from the kallsyms file to check for missing symbols. If some symbols are
 // missing, it will cancel their event with informative error message.
-func (t *Tracee) validateKallsymsDependencies() {
+func (t *Tracee) validateKallsymsDependencies(ps *policy.Policies) {
 	depsToCancel := make(map[events.ID]string)
 
 	// Cancel events with unavailable symbols dependencies
-	for eventToCancel, missingDepSyms := range t.getUnavKsymsPerEvtID() {
+	for eventToCancel, missingDepSyms := range t.getUnavKsymsPerEvtID(ps.EventsStates()) {
 		eventNameToCancel := events.Core.GetDefinitionByID(eventToCancel).GetName()
 		logger.Debugw(
 			"Event canceled because of missing kernel symbol dependency",
 			"missing symbols", missingDepSyms, "event", eventNameToCancel,
 		)
-		delete(t.eventsState, eventToCancel)
+		delete(ps.EventsStates().GetAll(), eventToCancel)
```

Comment:
This is temporary, none direct access to the map should occur. One more candidate to lie in policy pkg.